receiver/sqlserverreceiver/scraper.go

```diff
@@ -299,3 +321,254 @@ func (s *sqlServerScraperHelper) recordDatabaseStatusMetrics(ctx context.Context
 
 	return errors.Join(errs...)
 }
+
+func (s *sqlServerScraperHelper) recordDatabaseSampleQuery(ctx context.Context) (plog.Logs, error) {
+	const dbPrefix = "sqlserver."
+	// Constants are the column names of the database status
+	const DBName = "db_name"
+	const clientAddress = "client_address"
+	const clientPort = "client_port"
+	const queryStart = "query_start"
+	const sessionID = "session_id"
+	const sessionStatus = "session_status"
+	const requestStatus = "request_status"
+	const hostname = "host_name"
+	const command = "command"
+	const statementText = "statement_text"
+	const blockingSessionID = "blocking_session_id"
+	const waitType = "wait_type"
+	const waitTime = "wait_time"
+	const waitResource = "wait_resource"
+	const openTransactionCount = "open_transaction_count"
+	const transactionID = "transaction_id"
+	const percentComplete = "percent_complete"
+	const estimatedCompletionTime = "estimated_completion_time"
+	const cpuTime = "cpu_time"
+	const totalElapsedTime = "total_elapsed_time"
+	const reads = "reads"
+	const writes = "writes"
+	const logicalReads = "logical_reads"
+	const transactionIsolationLevel = "transaction_isolation_level"
+	const lockTimeout = "lock_timeout"
+	const deadlockPriority = "deadlock_priority"
+	const rowCount = "row_count"
+	const queryHash = "query_hash"
+	const queryPlanHash = "query_plan_hash"
+	const contextInfo = "context_info"
+
+	const username = "username"
+	rows, err := s.client.QueryRows(ctx)
+	if err != nil {
+		if errors.Is(err, sqlquery.ErrNullValueWarning) {
+			// TODO: ignore this for now.
+			s.logger.Warn("problems encountered getting log rows", zap.Error(err))
+		} else {
+			return plog.Logs{}, fmt.Errorf("sqlServerScraperHelper failed getting log rows: %w", err)
+		}
+	}
+
+	var errs []error
+	logs := plog.NewLogs()
+
+	resourceLog := logs.ResourceLogs().AppendEmpty()
+	resourceLog.Resource().Attributes().PutStr("db.system.type", "microsoft.sql_server")
+
+	scopedLog := resourceLog.ScopeLogs().AppendEmpty()
+	scopedLog.Scope().SetName("github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sqlserverreceiver")
+	scopedLog.Scope().SetVersion("v0.0.1")
+	for _, row := range rows {
+		queryHashVal := hex.EncodeToString([]byte(row[queryHash]))
+		queryPlanHashVal := hex.EncodeToString([]byte(row[queryPlanHash]))
+		contextInfoVal := hex.EncodeToString([]byte(row[contextInfo]))
+
+		cacheKey := queryHashVal + "-" + queryPlanHashVal
+
+		if _, ok := s.cache.Get(cacheKey); !ok {
```

Comment:
Do we need to maintain cache and have the logic of checking if the cacheKey exists? According to this logic, once we collected one sample for a given cacheKey, we never collect it again. The idea of samples is to get multiple samples for the same query to collect multiple instances of execution of a normalized query. Using cache prevents this.

Reply:
I agree. Every sample event is an individual event. I am not sure cache is needed here.

Reply:
removed